src/gui/midi.cpp

```diff
@@ -61,10 +61,10 @@ Bit8u MIDI_evt_len[256] = {
 
 MidiHandler * handler_list = 0;
 
-MidiHandler::MidiHandler(){
-	next = handler_list;
+MidiHandler::MidiHandler() : next(handler_list)
+{
 	handler_list = this;
-};
+}
```

Comment:
This (existing) design is so convoluted.. first we assign our member `next` various to whatever the global `list` pointer is set to; then we update the "list" with the address of the new hander... umm.. even though the "list" actually isn't a list, and actually just points to the most recent handler. 

Some kind of linked-list spread across a member variable and a global pointer. Argh. :broom: